dashboard/src/pages/[workspaceSlug]/[appSlug]/users/index.tsx

```diff
@@ -1,38 +1,379 @@
-            : currentApplication.hasuraGraphqlAdminSecret,
-      }}
-    >
-      <Container>
-        <UsersList />
+  /**
+   * This will change the `disabled` field in the user to its opposite.
+   */
+  async function handleBanUser(user: RemoteAppUser) {
+    const banUser = updateUser({
+      variables: {
+        id: user.id,
+        user: {
+          disabled: !user.disabled,
+        },
+      },
+    });
+
+    await toast.promise(
+      banUser,
+      {
+        loading: user.disabled ? 'Unbanning user...' : 'Banning user...',
+        success: user.disabled
+          ? 'User unbanned succesfully.'
+          : 'User banned succesfully',
+        error: user.disabled
+          ? 'Error trying to unban user.'
+          : 'Error trying to ban user.',
+      },
+      { ...toastStyleProps },
+    );
+    await refetchProjectUsers();
+  }
+
+  function handleDeleteUser(user: RemoteAppUser) {
+    openAlertDialog({
+      title: 'Delete User',
+      payload: (
+        <Text>
+          Are you sure you want to delete the &quot;
+          <strong>{user.displayName}</strong>&quot; user? This cannot be undone.
+        </Text>
+      ),
+      props: {
+        onPrimaryAction: async () => {
+          await toast.promise(
+            deleteUser({
+              variables: {
+                id: user.id,
+              },
+            }),
+            {
+              loading: 'Deleting user...',
+              success: 'User deleted successfully.',
+              error: 'An error occurred while trying to delete this user.',
+            },
+            toastStyleProps,
+          );
+
+          await refetchProjectUsers();
+        },
+        primaryButtonColor: 'error',
+        primaryButtonText: 'Delete',
+        titleProps: { className: 'mx-auto' },
+        PaperProps: { className: 'max-w-lg mx-auto' },
+      },
+    });
+  }
+
+  async function handleEditUser(
+    values: EditUserFormValues,
+    user: RemoteAppUser,
+  ) {
+    const updateUserMutationPromise = updateUser({
+      variables: {
+        id: user.id,
+        user: {
+          displayName: values.displayName,
+          email: values.email,
+          avatarUrl: values.avatarURL,
+          emailVerified: values.emailVerified,
+          defaultRole: values.defaultRole,
+          phoneNumber: values.phoneNumber,
+          phoneNumberVerified: values.phoneNumberVerified,
+          locale: values.locale,
+        },
+      },
+    });
+
+    await toast.promise(
+      updateUserMutationPromise,
+      {
+        loading: `Updating user's settings...`,
+        success: 'User settings updated successfully',
+        error: 'Failed to update user settings.',
+      },
+      { ...toastStyleProps },
+    );
+    await refetchProjectUsers();
+    closeDrawer();
+  }
+
+  function handleViewUser(user: RemoteAppUser) {
+    openDrawer('EDIT_USER', {
+      title: 'User Details',
+      payload: {
+        user,
+        onEditUser: handleEditUser,
```

Comment:
> We're better off sending an onSubmit handler that would close the drawer. Then mutations in the EditUserForm component could just simply call it to close the drawer or do perform "after effect".

This sounds good, especially the way that our refetch logic works given that we have two apollo client instances (and the query must use the same variables as the previous time it was run.) Thus, it would make more sense to add any such logic there.